### PR TITLE
Add hotplug cases into gluster disk

### PR DESCRIPTION
**Pull request by chunfuwen**
Orchestrate gluster_disk.cfg to scale different attach mode,e.g attach-disk

Signed-off-by: chunfuwen <chwen@redhat.com>

### CODE REVIEW

**Comment by chunfuwen** [comment]
@will-Do
@waynesun09

Please take a review.

**Comment by chunfuwen** [comment]
@balamuruhans 

Could you please have a look?

File: libvirt/tests/src/virtual_disks/virtual_disks_gluster.py [hunk 1]
```diff
@@ -189,7 +172,41 @@ def test_pmsuspend(vm_name):
...
+                vm.start()
+        else:
+            if not vm.is_dead():
+                vm.destroy()
+
+        #If gluster_disk is True,use attach_device.
+        attach_option = params.get("attach_option", "")
+        if gluster_disk:
+            cmd_result = virsh.attach_device(domainarg=vm_name, filearg=custom_disk.xml,
+                                             flagstr=attach_option,
```
Comment (balamuruhans) [inline libvirt/tests/src/virtual_disks/virtual_disks_gluster.py]: For coldplug I am seeing attach_option as --config but for hotplug we should use --live and I couldn't see the flagstr to set accordingly.

File: libvirt/tests/src/virtual_disks/virtual_disks_gluster.py [hunk 2]
```diff
@@ -189,7 +172,41 @@ def test_pmsuspend(vm_name):
...
+
+        # set domain options
+        if dom_iothreads:
+            try:
+                vmxml.iothreads = int(dom_iothreads)
+                vmxml.sync()
+            except ValueError:
+                # 'iothreads' may not invalid number in negative tests
+                logging.debug("Can't convert '%s' to integer type"
+                              % dom_iothreads)
```
Comment (balamuruhans) [inline libvirt/tests/src/virtual_disks/virtual_disks_gluster.py]:
```
logging.debug("Can't convert '%s' to integer type",
              dom_iothreads)
```

File: libvirt/tests/src/virtual_disks/virtual_disks_gluster.py [hunk 3]
```diff
@@ -174,6 +156,7 @@ def test_pmsuspend(vm_name):
 
     # Back up xml file.
     vmxml_backup = vm_xml.VMXML.new_from_inactive_dumpxml(vm_name)
+    vmxml = vm_xml.VMXML.new_from_inactive_dumpxml(vm_name)
```
Comment (balamuruhans) [inline libvirt/tests/src/virtual_disks/virtual_disks_gluster.py]: `vmxml = vmxml_backup.copy()`

File: libvirt/tests/src/virtual_disks/virtual_disks_gluster.py [hunk 4]
```diff
@@ -189,7 +172,41 @@ def test_pmsuspend(vm_name):
...
+                # 'iothreads' may not invalid number in negative tests
+                logging.debug("Can't convert '%s' to integer type"
+                              % dom_iothreads)
+        if default_pool:
+            disks_dev = vmxml.get_devices(device_type="disk")
+            for disk in disks_dev:
+                vmxml.del_device(disk)
+            vmxml.sync()
+
+        #If hot plug,start VM first,otherwise stop VM if running.
```
Comment (balamuruhans) [inline libvirt/tests/src/virtual_disks/virtual_disks_gluster.py]: `# If hotplug` space after #

File: libvirt/tests/src/virtual_disks/virtual_disks_gluster.py [hunk 5]
```diff
@@ -189,7 +172,41 @@ def test_pmsuspend(vm_name):
...
+        if start_vm:
+            if vm.is_dead():
+                vm.start()
+        else:
+            if not vm.is_dead():
+                vm.destroy()
+
+        #If gluster_disk is True,use attach_device.
```
Comment (balamuruhans) [inline libvirt/tests/src/virtual_disks/virtual_disks_gluster.py]: same here

File: libvirt/tests/src/virtual_disks/virtual_disks_gluster.py (hunk 1 above)
Comment (chunfuwen) [inline libvirt/tests/src/virtual_disks/virtual_disks_gluster.py]: For hot plug ,attach-device default option is --live ,so no need explicitly set this flag,I think.

File: libvirt/tests/src/virtual_disks/virtual_disks_gluster.py (hunk 3 above)
Comment (chunfuwen) [inline libvirt/tests/src/virtual_disks/virtual_disks_gluster.py]: done

File: libvirt/tests/src/virtual_disks/virtual_disks_gluster.py (hunk 2 above)
Comment (chunfuwen) [inline libvirt/tests/src/virtual_disks/virtual_disks_gluster.py]: Done

File: libvirt/tests/src/virtual_disks/virtual_disks_gluster.py (hunk 4 above)
Comment (chunfuwen) [inline libvirt/tests/src/virtual_disks/virtual_disks_gluster.py]: Done

File: libvirt/tests/src/virtual_disks/virtual_disks_gluster.py (hunk 5 above)
Comment (chunfuwen) [inline libvirt/tests/src/virtual_disks/virtual_disks_gluster.py]: Done

**Comment by chunfuwen** [comment]
@balamuruhans 
The new patch had been uploaded.

File: libvirt/tests/src/virtual_disks/virtual_disks_gluster.py [hunk 6]
```diff
@@ -189,7 +172,41 @@ def test_pmsuspend(vm_name):
...
+                # 'iothreads' may not invalid number in negative tests
+                logging.debug("Can't convert '%s' to integer type",
+                              dom_iothreads)
+        if default_pool:
+            disks_dev = vmxml.get_devices(device_type="disk")
+            for disk in disks_dev:
+                vmxml.del_device(disk)
+            vmxml.sync()
+
+        #If hot plug, start VM first, otherwise stop VM if running.
```
Comment (balamuruhans) [inline libvirt/tests/src/virtual_disks/virtual_disks_gluster.py]: missed this change to push

File: libvirt/tests/src/virtual_disks/virtual_disks_gluster.py [hunk 7]
```diff
@@ -189,7 +172,41 @@ def test_pmsuspend(vm_name):
...
+        if start_vm:
+            if vm.is_dead():
+                vm.start()
+        else:
+            if not vm.is_dead():
+                vm.destroy()
+
+        #If gluster_disk is True, use attach_device.
```
Comment (balamuruhans) [inline libvirt/tests/src/virtual_disks/virtual_disks_gluster.py]: same here

File: libvirt/tests/src/virtual_disks/virtual_disks_gluster.py (hunk 6 above)
Comment (chunfuwen) [inline libvirt/tests/src/virtual_disks/virtual_disks_gluster.py]: Updated

File: libvirt/tests/src/virtual_disks/virtual_disks_gluster.py (hunk 7 above)
Comment (chunfuwen) [inline libvirt/tests/src/virtual_disks/virtual_disks_gluster.py]: updated

**Comment by chunfuwen** [comment]
@balamuruhans ,
Thank you for reminding, new patch had been uploaded.

**Pull request merged**